src/core/index.ts

```diff
@@ -514,6 +515,22 @@ export class ThirdwebSDK implements IThirdwebSdk {
   }
 
   public async createSnapshot(leafs: string[]): Promise<SnapshotInfo> {
+    const counts: { [key: string]: any } = {};
+    const repeated: string[] = [];
+    leafs.forEach((leaf) => {
+      leaf = leaf.toLowerCase();
+      invariant(isAddress(leaf), "Invalid address : " + leaf);
+      if (counts[leaf] && repeated.indexOf(leaf) === -1) {
+        repeated.push(leaf);
+      } else {
+        counts[leaf] = 1;
+      }
+    });
+    invariant(
```

Comment:
Instead of using `invariant` here, lets throw custom error types so they're easier to catch.